### PR TITLE
Add setting to clear storage

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/1271

### CODE REVIEW

File: app/src/main/java/org/openobservatory/ooniprobe/activity/PreferenceActivity.java [hunk 1]
```diff
@@ -30,4 +43,27 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+		if (i == DialogInterface.BUTTON_POSITIVE) {
+			if (serializable.equals(R.id.delete)) {
+				CountlyManager.recordEvent("ClearStorage");
+				//From https://guides.codepath.com/android/using-dialogfragment
+				ProgressDialog pd = new ProgressDialog(this);
+				pd.setCancelable(false);
+				pd.show();
+				Result.deleteAll(this);
+				pd.dismiss();
+                fragment.setStorage();
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/PreferenceActivity.java:65]:
```suggestion
				fragment.setStorage();
```

File: app/src/main/java/org/openobservatory/ooniprobe/activity/PreferenceActivity.java [hunk 2]
```diff
@@ -30,4 +43,27 @@ public boolean onOptionsItemSelected(MenuItem item) {
 				return super.onOptionsItemSelected(item);
 		}
 	}
+	public void removeAllTests(View view){
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/activity/PreferenceActivity.java:46]:
```suggestion

	public void removeAllTests(View view){
```

File: app/src/main/java/org/openobservatory/ooniprobe/fragment/ResultListFragment.java [hunk 3]
```diff
@@ -154,7 +154,6 @@ public boolean onOptionsItemSelected(MenuItem item) {
                         .withMessage(getString(R.string.Modal_DoYouWantToDeleteAllTests))
                         .withPositiveButton(getString(R.string.Modal_Delete))
                         .build().show(getChildFragmentManager(), null);
-                CountlyManager.recordEvent("DeleteAllTests");
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/fragment/ResultListFragment.java:157]: Why was this moved?

File: app/src/main/java/org/openobservatory/ooniprobe/model/database/Measurement.java [hunk 4]
```diff
@@ -179,6 +179,26 @@ public void deleteLogFileAfterAWeek(Context c) {
 		}
 	}
 
+	public static long getStorageUsed(Context c){
+		String database = AppDatabase.NAME + ".db";
+		return getFolderSize(getMeasurementDir(c)) +
+				c.getDatabasePath(database).length() +
+				c.getDatabasePath(database + "-shm").length() +
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/model/database/Measurement.java:186]: What is the content of the `-shm` file? Should we count it?

**Review by bassosimone — COMMENTED** [review]
Seems good but I'd rather ask about two places where I'm unsure!

🧞‍♂️

File: app/src/main/java/org/openobservatory/ooniprobe/model/database/Measurement.java (hunk 4 above)
Comment (lorenzoPrimi) [reply app/src/main/java/org/openobservatory/ooniprobe/model/database/Measurement.java:186]: The WAL-Index File Format. The WAL-index or "shm" file is used to coordinate access to the database by multiple clients, and as a cache to help clients quickly locate frames within the wal file.

Comment (bassosimone) [reply app/src/main/java/org/openobservatory/ooniprobe/model/database/Measurement.java:186]: 👍

File: app/src/main/java/org/openobservatory/ooniprobe/fragment/ResultListFragment.java (hunk 3 above)
Comment (lorenzoPrimi) [reply app/src/main/java/org/openobservatory/ooniprobe/fragment/ResultListFragment.java:157]: Because we want to record how many people actually delete the tests, not the ones who just press on the button (before modal)

**Review by bassosimone — APPROVED** [review]
🐳
